[Feat] #7 place 사용자 위치 기준으로 조회하는 기능을 구현했어요

dong-yxxn opened the pull request:
### ✨ 작업 내용

- 코드 리뷰 반영했습니다.
- 특정 위치 중심으로 장소 조회 기능 구현
- 패키지명 수정 및 구조 재배치
---

### ✨ 참고 사항
- likes, influencer, category로 필터링 기능은 추후 구현 예정입니다.
- 테스트 코드 구현 완료하였습니다.
---

### ⏰ 현재 버그

---

### ✏ Git Close

suhyeon7497 committed:
chore: security 관련 dependency 추가

oauth, security와 관련된 dependency를 추가하였습니다.

관련 이슈: #2

suhyeon7497 committed:
feat: user 정보 저장을 위한 요소 추가

UserEntity와 Repository를 추가 하였습니다.

관련 이슈: #2

suhyeon7497 committed:
feat: oauth 로그인을 위한 security config 설정

oauth 로그인을 위해
csrf를 차단 (rest api이기 때문에 jwt로 검증을 함),
formlogin을 차단(사용하지 않음),
httpBasic을 차단(사용하지 않음) 하였습니다.
session에 stateless 속성을 부여하였습니다.

관련 이슈: #2

suhyeon7497 committed:
chore: 민감한 환경변수를 위한 서브 모듈 추가

서브 모듈을 추가하여 민감한 정보를 관리합니다.

관련 이슈:#2

suhyeon7497 committed:
chore: 서브모듈 최신화

서브 모듈을 최신화 하여
api 키를 업데이트하였습니다

관련 이슈: #2

suhyeon7497 committed:
chore: application.properties 인식 문제 해결

.gitignore 파일에 application.properties를 넣지안았는데도
추적되지 않음. 그 문제를 해결

관련 이슈: #2

suhyeon7497 committed:
feat: authentication 구현

authentication을 구현하였습니다.
kakao만 한다는 가정하에 kakao를 직접 넣었습니다.

관련 이슈: #2

Commit:
style : 컨벤션에 맞게 도메인별 패키지 사용

더미 데이터여서 그냥 만들었었는데, 피드백에 따라 변경하기로 결정

Commit:
style : 컨벤션에 맞게 패키지명 변경

패키지 명은 모두 소문자로 한다는 컨벤션에 맞추어 소문자로 변경

Commit:
style : 생성자 lombok으로 변경

Commit:
style : 생성자 lombok으로 변경

Commit:
chore : 주석 추가 ( 클래스 설명 )

Commit:
Merge branch 'weekly/1' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#5-videoAlias

Commit:
refactor : Pull 이후에 사용하던 더미 데이터 삭제

Commit:
refactor : Pull 이후에 사용하던 더미 데이터 부분을 실제 값으로 대체

Commit:
feat : videoAlias 정보를 랜덤하게 매핑해주는 기능 구현

해당 기능 최초 구현, 구현 방식이나 데이터의 추가가 개선되어야 할 것 같다

dong-yxxn committed:
[refactor]: 코드리뷰 반영

- PlaceService 인터페이스 제거, getCategories()에서 List 반환
- Place id의 nullable=false 제거
- CategoryListDTO -> CategoryListResponse로 이름 변경해서 직관화함

관련 이슈: #7

dong-yxxn committed:
[refactor]: 코드리뷰 반영

- PlaceService 인터페이스 제거, getCategories()에서 List 반환
- Place id의 nullable=false 제거
- CategoryListDTO -> CategoryListResponse로 이름 변경해서 직관화함

관련 이슈: #7

Commit:
style : 변수 선언 시점 변경

사용 지점과 최대한 가깝게 변수 선언 시점을 변경

Commit:
style : savedInfluencers 변수 삭제

메서드 체이닝을 통해 필요 없는 변수를 삭제

dong-yxxn committed:
[feat] PlaceTime, Menu VO를 Place 클래스에 추가

PlaceTime VO와 Menu VO를 Place 클래스에 추가후, 테스트 코드까지 수정하였습니다.

관련 이슈: #7

dong-yxxn committed:
[feat] PlaceTime, Menu VO를 Place 클래스에 추가

PlaceTime VO와 Menu VO를 Place 클래스에 추가후, 테스트 코드까지 수정하였습니다.

관련 이슈: #7

Commit:
refactor : 테스트를 위한 생성자 추가 및 필드 변수 설정

id를 제외한 필드를 사용하여 생성자를 만들고, 이를 통해 테스트를 진행하기 위하여 @NonNull과 @RequiredArgsConstructor을 사용하여 리펙토링

Commit:
[test] 비디오 도메인에 대한 테스트 코드 작성

추후에 통합 테스트 필요 ( + POSTMAN을 이용한 API 테스트도 필요 )

Commit:
test : 비디오 도메인에 대한 테스트 코드 작성

추후에 통합 테스트 필요 ( + POSTMAN을 이용한 API 테스트도 필요 )

suhyeon7497 committed:
fix: authentication 하면 무한 리다이렉트 해결

/login 에 authenicated가 적용되어 있어 무한 리다이렉트 되던 문제를
해결하였습니다.

관련 이슈:#2

suhyeon7497 committed:
feat: oauth 로그인 하면 유저 정보 저장

oauth 로그인 시 유저 정보를 db에 저장

관련 이슈: #2

dong-yxxn committed:
[refactor] 패키지명 수정

패키지를 application-domain-persistence-presentaion 구조로 설정하였습니다.
List<Category> 대신, CategoryInfo를 구현해서 dto로 적용시켰습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor] 패키지명 수정

패키지를 application-domain-persistence-presentaion 구조로 설정하였습니다.
List<Category> 대신, CategoryInfo를 구현해서 dto로 적용시켰습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor] Address, Coordinate를 VO로 설정

Place 클래스에서 주소와 위도, 경도를 각각 Address와 Coordinate VO로 추출했습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor] Address, Coordinate를 VO로 설정

Place 클래스에서 주소와 위도, 경도를 각각 Address와 Coordinate VO로 추출했습니다.

관련 이슈: #7

Commit:
[refactor] 9/23 회의 내용 반영

record 컨벤션, dto 컨벤션 반영 및 패키지 구조 변경

Commit:
Merge remote-tracking branch 'origin/feat/#5-videoAlias' into feat/#5-videoAlias

# Conflicts:
#	src/test/java/team7/inplace/VideoTest/domain/VideoTest.java
#	src/test/java/team7/inplace/VideoTest/repository/VideoRepositoryTest.java
#	src/test/java/team7/inplace/VideoTest/service/VideoServiceTest.java

suhyeon7497 committed:
[chore]submodule 제거

submodule이 내 private repository라서 사용하지 않기로 정함
고로 삭제

관련 이슈: #2

suhyeon7497 committed:
Merge branch 'weekly/1' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#2-authentication

suhyeon7497 committed:
[chore] 패키지 구조 변경

9/23일자로 변경된 패키지 컨벤션으로 패키지 구조를 변경하였습니다.

관련 이슈: #2

suhyeon7497 committed:
[fix] security 설정 없으면 오류 발생 해결

security에 대한 설정이 없으면 애플리케이션 실행 안돼서
application.yml 파일 추가

관련 이슈: #2

Commit:
[refactor] 피드백 내용 반영

랜덤한 Alias를 매핑해주는 기능은 Service의 영역이 아니라는 피드백을 수용하여 이를 분리

dong-yxxn committed:
[feat] 사용자 기준으로 장소 조회 기능 구현

- DataLoader로 더미데이터 저장
- Controller: 사용자 위치 기준으로 장소 조회
- Service, Repo: 가까운순으로 장소 조회 로직 구현
- PlaceComm: 위도 경도, page 정보를 service단으로 전송하는 dto
- PlaceInfo: 조회된 장소 정보를 controller 단으로 전송하는 dto

관련 이슈: #7

dong-yxxn committed:
[feat] 사용자 기준으로 장소 조회 기능 구현

- DataLoader로 더미데이터 저장
- Controller: 사용자 위치 기준으로 장소 조회
- Service, Repo: 가까운순으로 장소 조회 로직 구현
- PlaceComm: 위도 경도, page 정보를 service단으로 전송하는 dto
- PlaceInfo: 조회된 장소 정보를 controller 단으로 전송하는 dto

관련 이슈: #7

dong-yxxn committed:
Merge remote-tracking branch 'origin/develop' into feat/#7-place

suhyeon7497 committed:
[chore] jwt 관련 dependency 추가

jwt 토큰 발급을 위해 dependency를 추가 하였습니다.

관련 이슈: #2

suhyeon7497 committed:
[feat] jwt를 발급, 검증하는 util 클래스

jwt토큰을 발급하고 검증하는 JWTUtil 클래스를 작성하였습니다.

관련 이슈: #2

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -0,0 +1,74 @@
+    public Page<PlaceInfo> getPlacesWithinRadius(
+        PlacesCoordinateCommand placesCoordinateCommand) {
+
+        // 주어진 좌표로 장소를 찾고, 해당 페이지의 결과를 가져옵니다.
+        Page<Place> placesPage = getPlacesByDistance(placesCoordinateCommand);
+
+        // Place 객체를 PlaceInfo로 변환합니다.
+        List<PlaceInfo> placeInfoList = placesPage.getContent().stream()
+            .map(Place::getPlaceInfo)
+            .toList();
+
+        // PageImpl을 사용하여 Page<PlaceInfo>로 감쌉니다.
+        return new PageImpl<>(placeInfoList, placesPage.getPageable(),
+            placesPage.getTotalElements());
```
`Page<Place> placesPage = getPlacesByDistance(placesCoordinateCommand);`
`return placesPage.map(PlaceInfo::new);`
같은 느낌으로 하면 더 깔끔해질 것 같습니당

BaeJunH0 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -0,0 +1,74 @@
+            comm.pageable());
+    }
+
+    public List<CategoryInfo> getCategories() {
+        return Arrays.stream(Category.values()).map(category -> new CategoryInfo(category.name()))
+            .toList();
+    }
```
카테고리는 따로 처리해야하지 않을까요? 서비스의 역할이 아닌 것 같아용

sanghee0820 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -0,0 +1,74 @@
+
+    public PlaceService(PlaceRepository placeRepository) {
+        this.placeRepository = placeRepository;
+    }
```
@RequiredArgsConstructor 사용 하시면 어떠실까요?

dong-yxxn committed:
[refactor] Category 관련 로직을 CategoryService로 구현

- Place의 id를 placeId -> id로 변경
- getCategories() 로직을 CategoryService구현해서 분리

관련 이슈: #7

dong-yxxn committed:
[refactor] Category 관련 로직을 CategoryService로 구현

- Place의 id를 placeId -> id로 변경
- getCategories() 로직을 CategoryService구현해서 분리

관련 이슈: #7

BaeJunH0 commented on src/main/java/team7/inplace/place/domain/Place.java:
```diff
@@ -54,26 +60,40 @@ public class Place {
+    @CollectionTable(name = "menus", joinColumns = @JoinColumn(name = "place_id"))
+    private List<Menu> menuList;
+
+    public PlaceInfo getPlaceInfo() {
+        AddressInfo addressInfo = address.getAddressInfo();
+
+        return new PlaceInfo(
+            id,
+            name,
+            addressInfo,
+            category.toString(),
+            null,
+            coordinate.getLongitude(),
+            coordinate.getLatitude(),
+            null);
+    }
```
DTO를 생성해주는 부분이 비즈니스 로직인지 잘 모르겠습니다, DTO 생성자에서 처리하는 것이 맞지 않을까요?

sanghee0820 commented on src/main/java/team7/inplace/place/application/PlaceService.java:
```diff
@@ -0,0 +1,64 @@
+
+            AddressInfo addressInfo = place.getAddress().getAddressInfo();
+
+            CoordinateInfo coordinateInfo = place.getCoordinate().getCoordinateInfo();
+
+            PlaceInfo placeInfo = new PlaceInfo(placeId,
+                place.getName(),
+                addressInfo,
+                place.getCategory().toString(),
+                coordinateInfo);
+            placeInfos.add(placeInfo);
+        }
+        return placeInfos;
+    }
+*/
+
+    private Page<Place> getPlacesByDistance(PlacesCoordinateCommand comm) {
+        return placeRepository.getPlacesByDistance(
+            comm.latitude(),
+            comm.longitude(),
+            comm.pageable());
+    }
+
```
필요없는 주석 지워주세요

dong-yxxn committed:
[refactor] Place DTO 생성자 PlaceInfo로 이동

- PlaceInfo 생성자를 Entity말고 PlaceInfo 생성자로 이동시킴
- PlaceService: getPlacesWithinRadius에서 return값 단순화했습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor] Place DTO 생성자 PlaceInfo로 이동

- PlaceInfo 생성자를 Entity말고 PlaceInfo 생성자로 이동시킴
- PlaceService: getPlacesWithinRadius에서 return값 단순화했습니다.

관련 이슈: #7

BaeJunH0 reviewed (approved):
열심히 하셨습니다 승인할게요~

dong-yxxn committed:
[refactor]: PlaceInfo 팩토리 메서드로 변경함

- PlaceInfo 생성자를 Place에서 생성하도록 변경하였습니다.

관련 이슈: #7

dong-yxxn committed:
[refactor]: PlaceInfo 팩토리 메서드로 변경함

- PlaceInfo 생성자를 Place에서 생성하도록 변경하였습니다.

관련 이슈: #7

suhyeon7497 committed:
[refactor] code review 피드백 반영

코드 리뷰 피드백을 반영하였습니다.

관련 이슈: #2

sanghee0820 commented on src/main/java/team7/inplace/place/application/command/PlacesCommand.java:
```diff
@@ -0,0 +1,12 @@
+
+import org.springframework.data.domain.Pageable;
+
+public record PlacesCommand() {
+
+    public record PlacesCoordinateCommand(String longitude,
+                                          String latitude,
+                                          Pageable pageable) {
+
+    }
+}
```
```java
public class PlacesCommand {

    public record PlacesCoordinateCommand(
        String longitude,
        String latitude,
        Pageable pageable
    ) {
    }
}
```

이렇게 하는건 어떨까요?

sanghee0820 commented on src/main/java/team7/inplace/place/application/dto/PlaceInfo.java:
```diff
@@ -0,0 +1,22 @@
+package team7.inplace.place.application.dto;
+
+
+public record PlaceInfo(Long placeId,
+                        String placeName,
+                        AddressInfo address,
+                        String category,
+                        String influencerName,
+                        String longitude,
+                        String latitude,
+                        Boolean likes) {
+
+
+    // influencer, likes 추가 예정
+    public record AddressInfo(
+        String address1,
+        String address2,
+        String address3
```
이 부분도 포메팅 일치가 필요할 듯 싶습니다

sanghee0820 commented on src/main/java/team7/inplace/place/domain/Address.java:
```diff
@@ -0,0 +1,30 @@
+import team7.inplace.place.application.dto.PlaceInfo;
+
+@Embeddable
+@NoArgsConstructor
```
접근자 추가해주세요!

dong-yxxn committed:
[refactor] PlaceInfo 팩토리메서드 이름 of 및 static으로 변경

- PlaceInfo 팩토리메서드 이름을 of로 변경함
- 그 후 static 메서드로 지정

관련 이슈: #7

dong-yxxn committed:
[refactor] PlaceInfo 팩토리메서드 이름 of 및 static으로 변경

- PlaceInfo 팩토리메서드 이름을 of로 변경함
- 그 후 static 메서드로 지정

관련 이슈: #7

sanghee0820 commented on src/main/java/team7/inplace/place/domain/Address.java:
```diff
@@ -0,0 +1,30 @@
+    @Column(nullable = false, length = 50)
+    private String address3;
+
+    public PlaceInfo.AddressInfo getAddressInfo() {
+        return new PlaceInfo.AddressInfo(address1, address2, address3);
+    }
```
domain에서 dto를 만들게 되면, PlaceInfo가 변경되었을 때 domain을 수정하게 됩니다!

domain에서 외부 인자를 받는걸 지양해서 코드 작성해주시면 좋을 것 같아요

sanghee0820 commented on src/main/java/team7/inplace/place/persistence/PlaceRepository.java:
```diff
@@ -0,0 +1,30 @@
+public interface PlaceRepository extends JpaRepository<Place, Long> {
+
+    //거리 계산
+    @Query(value = "SELECT *, " +
+        "(6371 * acos(cos(radians(CAST(:latitude AS DECIMAL(10, 6)))) " +
+        "* cos(radians(CAST(latitude AS DECIMAL(10, 6)))) " +
+        "* cos(radians(CAST(longitude AS DECIMAL(10, 6))) - radians(CAST(:longitude AS DECIMAL(10, 6)))) "
+        +
+        "+ sin(radians(CAST(:latitude AS DECIMAL(10, 6)))) " +
+        "* sin(radians(CAST(latitude AS DECIMAL(10, 6)))))) AS distance " +
+        "FROM places " +
+        "ORDER BY distance",
+        countQuery = "SELECT count(*) FROM place",  // 총 개수 쿼리
+        nativeQuery = true)
```
나중에 QueryDsl을 사용해보는것도 좋을 것 같습니다.

sanghee0820 commented on src/main/java/team7/inplace/place/presentation/PlaceController.java:
```diff
@@ -0,0 +1,57 @@
+public class PlaceController {
+
+    private final PlaceService placeService;
+    private final CategoryService categoryService;
+
+    @GetMapping
+    public ResponseEntity<PlacesResponse> getPlaces(
+        @RequestParam String longitude,
+        @RequestParam String latitude,
+        @RequestParam(defaultValue = "0") int page,
+        @RequestParam(defaultValue = "10") int size,
+        @SortDefault.SortDefaults({
```
필요한 파라미터들을 클래스로 만들고, ModelAttribute로 받으면 좋을 것 같습니다.

sanghee0820 reviewed (changes requested):
리뷰 몇개 남겼습니다!

중복된 리뷰는 작성하지 않았으니 참고해서 수정 부탁드립니다!

dong-yxxn committed:
[refactor]: 코드리뷰 반영

- PlaceController: getPlaces() 파라미터를 클래스로 만들고 ModelAttribute로 받도록 하였습니다.
- Address: AddressInfo 생성 메서드를 AddressInfo 자체 메서드로 이동하였습니다.
- 그외: 포메팅 일치

관련 이슈: #7

dong-yxxn committed:
[refactor]: 코드리뷰 반영

- PlaceController: getPlaces() 파라미터를 클래스로 만들고 ModelAttribute로 받도록 하였습니다.
- Address: AddressInfo 생성 메서드를 AddressInfo 자체 메서드로 이동하였습니다.
- 그외: 포메팅 일치

관련 이슈: #7

Commit:
[refactor] 피드백 내용 반영

랜덤한 Alias를 매핑하는 기능을 Util 클래스와 Enum 클래스로 나눠 구현

Commit:
[refactor] 피드백 내용 반영

DTO 클래스의 static factory method 구현

suhyeon7497 committed:
[feat] : cookie를 사용해 token 반환

cookie를 통해 accessToken과 refreshToken을 반환하였습니다.
현재는 localhost:8080/successCookie로 redirect 하도록 했습니다.

관련 이슈: #2

sanghee0820 committed:
Merge pull request #16 from kakao-tech-campus-2nd-step3/feat/#5-videoAlias

[Feat] #5 videoAlias 정보를 랜덤하게 매핑해주는 기능을 구현했어요!

suhyeon7497 committed:
Merge branch 'weekly/2' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#2-authentication

sanghee0820 committed:
Merge pull request #17 from kakao-tech-campus-2nd-step3/feat/#2-authentication

[feat] #2 Oauth2 로그인 authentication을 구현하였습니다.

sanghee0820 committed:
Merge branch 'feat/#7-place' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#7-place

sanghee0820 committed:
[refactor] 매서드 이름 수정

sanghee0820 committed:
[refactor] import문 최적화

sanghee0820 reviewed (approved):
LGTM
